files/scripts/selinux/trivalent/trivalent.te

```diff
@@ -0,0 +1,249 @@
+allow trivalent_t user_home_t:lnk_file { create open read write rename unlink setattr getattr};
+allow trivalent_t audio_home_t:dir { add_name read remove_name watch write search getattr };
+allow trivalent_t audio_home_t:file { create getattr open read rename setattr unlink write };
+allow trivalent_t audio_home_t:lnk_file { create getattr open read rename setattr unlink write };
+
+
+
+# xwayland/nvidia
+xserver_exec(trivalent_t)
+dev_rw_xserver_misc(trivalent_t)
+dev_map_xserver_misc(trivalent_t)
+allow trivalent_t xserver_misc_device_t:chr_file { getattr ioctl map open read write };
+
+
+files_list_home(trivalent_t)
+files_search_home(trivalent_t)
+files_read_usr_files(trivalent_t)
+files_read_etc_files(trivalent_t)
+files_read_etc_runtime_files(trivalent_t)
+files_watch_etc_dirs(trivalent_t)
+files_getattr_all_dirs(trivalent_t)
+files_watch_root_dirs(trivalent_t)
+files_read_var_lib_files(trivalent_t)
+files_rw_generic_tmp_dir(trivalent_t)
+files_manage_generic_tmp_dirs(trivalent_t)
+files_rw_generic_tmp_sockets(trivalent_t)
+files_rw_tmp_file_leaks(trivalent_t)
+
+kernel_read_system_state(trivalent_t)
+kernel_read_kernel_sysctls(trivalent_t)
+kernel_read_fs_sysctls(trivalent_t)
+
+# required to connect to wayland
+unconfined_stream_connect(trivalent_t)
+dbus_system_bus_client(trivalent_t)
+dbus_session_bus_client(trivalent_t)
+
+dbus_write_session_tmp_sock_files(trivalent_t)
+devicekit_dbus_chat_disk(trivalent_t)
+devicekit_dbus_chat_power(trivalent_t)
+systemd_dbus_chat_hostnamed(trivalent_t)
+
+fs_rw_inherited_tmpfs_files(trivalent_t)
+fs_getattr_xattr_fs(trivalent_t)
+fs_getattr_tmpfs(trivalent_t)
+fs_rw_tmpfs_files(trivalent_t)
+fs_search_cgroup_dirs(trivalent_t)
+fs_associate_proc(trivalent_t)
+
+miscfiles_read_all_certs(trivalent_t)
+miscfiles_map_generic_certs(trivalent_t)
+miscfiles_read_localization(trivalent_t)
+miscfiles_watch_localization_dirs(trivalent_t)
+miscfiles_read_hwdata(trivalent_t)
+
+alsa_read_rw_config(trivalent_t)
+pulseaudio_tmpfs_content(trivalent_t)
+pulseaudio_stream_connect(trivalent_t)
+pulseaudio_read_home_files(trivalent_t)
+cups_read_config(trivalent_t)
+cups_stream_connect(trivalent_t)
+
+dev_read_sysfs(trivalent_t)
+dev_rw_dma_dev(trivalent_t)
+dev_rw_dri(trivalent_t)
+dev_rw_generic_usb_dev(trivalent_t)
+dev_read_video_dev(trivalent_t)
+dev_read_sound(trivalent_t)
+dev_write_sound(trivalent_t)
+dev_read_urand(trivalent_t)
+dev_read_rand(trivalent_t)
+
+udev_read_pid_files(trivalent_t)
+
+gnome_search_gconf_data_dir(trivalent_t)
+gnome_manage_cache_home_dir(trivalent_t)
+gnome_manage_generic_cache_files(trivalent_t)
+gnome_manage_generic_cache_sockets(trivalent_t)
+gnome_map_generic_cache_files(trivalent_t)
+gnome_manage_home_config(trivalent_t)
+gnome_manage_home_config_dirs(trivalent_t)
+gnome_manage_data(trivalent_t)
+gnome_map_generic_data_home_files(trivalent_t)
+gnome_manage_gstreamer_home_files(trivalent_t)
+gnome_dbus_chat_gconfdefault(trivalent_t)
+gnome_dbus_chat_gkeyringd(trivalent_t)
```

Comment:
Is there any reason this isn't all optional policy? Non gnome images probably don't have any of these interfaces